packages/dd-trace/src/proxy.js

```diff
@@ -33,6 +27,21 @@ class LazyModule {
   }
 }
 
+function lazyProxy (obj, property, config, getClass, ...args) {
+  const get = () => {
+    const RealClass = getClass()
+    const value = new RealClass(...args)
+
+    Reflect.defineProperty(obj, property, { value, configurable: true, enumerable: true })
+
+    return value
+  }
+
+  if (config?._isInServerlessEnvironment?.() === false) return get()
```

Comment:
So wait, if we're not in severless, we return a value, otherwise we just define a property? That seems odd.

Reply:
The value doesn't need to be returned, I just wanted to return early and returning the statement makes it a one-liner. I can split this into 4 lines if you think it makes the code easier to understand. The important bits is to call `get()` so that the prototype is overridden and then return early.

Reply:
Refactored the code to make the distinction clearer, remove the early return, and no longer return a value unnecessarily.